### ISSUE TITLE
Add support for ppc64le architecture

### ISSUE THREAD

**Issue by synapticarbors** [microsoft/LightGBM#4736]
## Description
When building `lightgbm` in conda-forge for linux ppc64le tests derived from https://github.com/microsoft/LightGBM/blob/master/tests/python_package_test/test_sklearn.py are failing. The exact code is in:

https://github.com/conda-forge/lightgbm-feedstock/blob/09db91759a5947555acb7de03d7ac08a527b3821/recipe/run_test.py

## Reproducible example
See:

https://github.com/conda-forge/lightgbm-feedstock/blob/09db91759a5947555acb7de03d7ac08a527b3821/recipe/run_test.py

This fails with

```python
======================================================================
FAIL: test_regression (__main__.TestSklearn)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/lightgbm_1635387498724/test_tmp/run_test.py", line 40, in test_regression
    self.assertLess(ret, 16)
AssertionError: 2.1663368127175135e+45 not less than 16

----------------------------------------------------------------------

```

## Environment info

release version 3.3.1 from pypi

Command(s) you used to install LightGBM

```shell

```

CI build log is at:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=398845&view=logs&jobId=bb1c2637-64c6-57bd-9ea6-93823b2df951&j=bb1c2637-64c6-57bd-9ea6-93823b2df951&t=350df31b-3291-5209-0bb7-031395f0baa1

**Comment by StrikerRUS** [comment]
This is subissue of #1129.

Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.

**Comment by jameslamb** [comment]
Sorry, this was locked accidentally. Just unlocked it. We'd still love help with this feature!
